src/components/App.js

```diff
@@ -1,30 +1,57 @@
-import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, IconButton, VisibilityOff, FormControl, FormLabel, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
+import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, FormControl, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
 import AddIcon from '@mui/icons-material/Add'
-import Visibility from '@mui/icons-material/Visibility'
 import React, { useState } from "react"
+import { useAppData } from './context/hooks'
+import { MUTATION_KEYS, useMutation } from '../config/queryClient'
 
 function App() {
+  const { data } = useAppData()
+  const { mutate: postAppData } = useMutation(MUTATION_KEYS.POST_APP_DATA)
+
   const [question, setQuestion] = useState("");
-  const [choice1, setChoice1] = useState("");
-  const [choice2, setChoice2] = useState("");
+  const [choices, setChoices] = useState(["", ""]);
 
-  var choiceList = []
-  choiceList.push([choice1, setChoice1])
-  choiceList.push([choice2, setChoice2])
+  const [answers, setAnswers] = React.useState([{
+    answer: false
+  },
+  {
+    answer: false
+  }]);
 
-  //const [answers, setAnswers] = useState([{ value: 'it is answer 1', checked: false }])
+  const handleAnswerCorrectnessChange = (index, e) => {
+    let newAnswers = [...answers]
+    newAnswers[index].answer = e.target.checked
+    setAnswers(newAnswers)
+  };
 
-  const [answers, setAnswers] = React.useState({
-    answer1: false,
-    answer2: false,
-  });
+  const handleChoiceChange = (index, e) => {
+    let newChoices = [...choices]
+    newChoices[index] = e.target.value
+    setChoices(newChoices)
+  }
 
-  const { answer1, answer2 } = answers;
+  const addAnswer = () => {
+    setChoices([...choices, ""])
+    setAnswers([...answers, { answer: false }])
```

Comment:
Could we factor out `""` and `{ answer: false }` as DEFAULT_CHOICE and DEFAULT_ANSWER? And reuse in line 12-19